src/Hamiltonians/FroehlichPolaron.jl

```diff
@@ -0,0 +1,159 @@
+"""
+function FroehlichPolaron(
+    addr::OccupationNumberFS;
+    v=1.0,
+    mass=1.0,
+    omega=1.0,
+    l=1.0,
+    p=0.0,
+    momentum_cutoff=nothing,
+    mode_cutoff=100.0,
+)
+
+The 1D Froehlich polaron Hamiltonian is given by
+
+```math
+H = \\frac{(P̂_f - P)^2}{m} + ωN̂ - ν Σₖ(âₖ^† + â_{-k})
+```
+
+where ``P`` is the total momentum `total_mom`, ``P̂_f = \\frac{M}{L}Σ_k k âₖ^† âₖ`` is the momentum operator for the bosons, ``N̂ = Σ_k âₖ^† âₖ`` is the number operator for the bosons, and ``ν`` is the coupling strength. Here, ``M`` is the number of discrete momentum modes and ```L`` is the momentum scaling.
```

Comment:
A single line should not be more than 92 characters long. See here https://github.com/JuliaDiff/BlueStyle for a style guide that we try to follow (approximately).

What is $L$? Is it the same as `l`? From the docstring it should become super clear how the model parameter, symbols in the formula, and the function arguments relate.

Is ``ν`` a Greek letter \nu? It might be better to use the Latin $v$, as function argument is called `v`.